### ISSUE TITLE
TryPerformCloseFragmentTransaction uses CommitAllowingStateLoss

### ISSUE THREAD

**Issue by softlion** [MvvmCross/MvvmCross#3265]
## 🐛 Bug Report

In `MvxAppCompatViewPresenter.cs`: 

the TryPerformCloseFragmentTransaction() method uses CommitAllowingStateLoss() to commit the fragment transaction. But it also may use PopBackStackImmediate to close the fragment. 

Why not use CommitNowAllowingStateLoss() instead to match the "immediate" version of PopBackStack ?
Or use the non immediate version of PopBackStack to match the non immediate version of CommitAllowingStateLoss ?

### Expected behavior

Coherence

**Version:** 6.2.2

**Platform:** 
- [ ] :iphone: iOS
- [X] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by tbalcom** [comment]
Changing the AppCompat presenter to use `CommitNowAllowingStateLoss()` in `PerformShowFragmentTransaction(FragmentManager fragmentManager, MvxFragmentPresentationAttribute attribute, MvxViewModelRequest request)` causes [IllegalStateException: This transaction is already being added to the backstack](https://stackoverflow.com/a/41051627). 

The error message is kind of deceptive, it's really saying you can't use `FragmentTransaction.AddToBackStack()` with `CommitNowAllowingStateLoss()`. This doesn't seem worth implementing for a majority of cases.

Some [more information](https://medium.com/@bherbst/the-many-flavors-of-commit-186608a015b1) is available:

> The caveat is that you can’t use commitNow() with a transaction that you are adding to the back stack. Think about it this way- if you were to add a transaction to the back stack via commit() then immediately add a transaction to the back stack via commitNow(), what does the back stack look like? Because the framework can’t provide any guarantees regarding the ordering here, it simply isn’t supported.

**Comment by nmilcoff** [comment]
As far as I understand, the Commit / Pop methods do not match in pairs.

In other words, `AllowingStateLoss` and `Immediate` are not related.

Am I wrong there?

**Comment by softlion** [comment]
Exactly. But using Now requires a good state of fragments. Not using Now defer the action to when the state permits it. This may be an issue with await navigation methods where the task returns sooner than it should.
